### DISCUSSION TITLE
HA Energy Panel Incorrect

### DISCUSSION THREAD

**Discussion by manuelsirianni** [macxq/foxess-ha#197] [Q&A]
Hi, i've installed the integration and i can get the values in the FoxESS Entities.
I've tried to configure the Energy Panel but it seems to have some issues because it's not reflecting what i can see (considering delay) on the FoxCloud app and the flows are really strange because they are going in all directions (unless Grid to Battery).

I've used the following fields:
Grid (Consumed from Network) --> Grid Consumption
Grid (Back to Network) --> Feedin
Solar Panels (Solar Production) --> Energy Generated
Domestic Battery --> Bat Discharge + Bat Charge

There something wrong?
I'm getting updated data in all entities every 5 minutes as expected.
I've only one string, so i do not need to sum PV1, PV2, etc..

Thanks!
Regards

**Comment by manuelsirianni** [comment]


**Reply by Daibutt** [reply]
If you have one string you may still need PV1 sum... without this the HA energy dashboard will not know what energy is generated from your PV and will make assumptions about where that power is coming from (i.e. possibly making your house load appear as negative).

It's a while since I've used this integration but I believe energy generated includes battery discharge. Is there a sensor called solar generation? You could try using that for solar entry. If not try PV1 sum.

https://github.com/macxq/foxess-ha/wiki/Understand-PV-string-power-generation-using-foxess-ha
